.github/workflows/node.js.yml

```diff
@@ -0,0 +1,44 @@
+# This workflow will do a clean install of node dependencies, build the source code and run tests across different versions of node
```

Comment:
Maybe not neccessary. We have appveyor for windows build, you can add windows related there.

Reply:
@gengjiawen We do, and it ran into an exception that was thrown (due to a bug in my code), yet the pipeline was marked as passing :(
https://ci.appveyor.com/project/tabrindle/envinfo/builds/34224373#L898

I'm not super familiar with appveyor (we use github actions and azure devops at microsoft 🙂) but I've verified that in the GH actions workflow I'm adding, a thrown exception will make the pipeline fail and block merging a PR

Reply:
We're just swallowing exceptions:
https://ci.appveyor.com/project/tabrindle/envinfo/builds/34224373#L26
```
node -e "try{require('./postinstall')}catch(e){}"
```

Reply:
Maybe try adding `--unhandled-rejections=strict` https://github.com/tabrindle/envinfo/blob/369259a458954ef3577063991e87750ce606609a/appveyor.yml#L18

Reply:
@gengjiawen thanks, yes that's what I figured I had to add in github actions :) I'll add it to appveyor too.
I think we should have these sanity runs on different services is that the environments they will be running will be different - some might have some browsers/ide/sdks, some might not, etc. Let me know if you disagree and still want me to remove the github one.
Is there any pipeline that is testing on ubuntu?

Reply:
@gengjiawen I added it to the appveyor pipeline but that one is running with node 8.0, and `--unhandled-rejections` was added on node 12.0 so I think we still want the gha running for extra coverage